src/main/java/com/github/havardh/javaflow/phases/resolver/FileResolver.java

```diff
@@ -0,0 +1,99 @@
+package com.github.havardh.javaflow.phases.resolver;
+
+import com.github.havardh.javaflow.exceptions.ExitException;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Resolver to resolve a single pattern into a list of matching files
+ */
+public class FileResolver {
+
+  /**
+   * Reads a single pattern into a list of matching files
+   * The patterns can either be a relative or absolute path
+   * to a file or directory. Or a relative or absolute glob.
+   * All patterns are relative to the working directory of
+   * the process.
+   *
+   * @param pattern the file, directory or glob pattern
+   * @return a list of paths matching the pattern
+   */
+  public List<Path> resolve(String pattern) {
+    try {
+      Path path = Paths.get(".").resolve(pattern);
+      if (Files.isRegularFile(path)) {
+        return normalize(Collections.singletonList(path));
+      } else if (Files.isDirectory(path)) {
+        return normalize(ofDirectory(path));
+      } else {
+        return normalize(ofPattern(pattern));
+      }
+    } catch (IOException e) {
+      throw new ExitException(ExitException.ErrorCode.COULD_NOT_RESOLVE_PATTERN, e);
+    }
+  }
+
+  private static List<Path> ofDirectory(Path path) throws IOException {
+    FileVisitor visitor = FileVisitor.matchAllFiles();
+    Files.walkFileTree(Paths.get(".").resolve(path), visitor);
+    return visitor.getPaths();
+  }
+
+  private static List<Path> ofPattern(String pattern) throws IOException {
+    FileVisitor visitor = FileVisitor.matchingGlob(pattern);
+    Files.walkFileTree(Paths.get("."), visitor);
+    return visitor.getPaths();
+  }
+
+  private static List<Path> normalize(List<Path> paths) {
+    return paths.stream().map(Path::normalize).collect(Collectors.toList());
+  }
+
+  private static final class FileVisitor extends SimpleFileVisitor<Path> {
+
+    private final PathMatcher pathMatcher;
+    private final List<Path> paths = new ArrayList<>();
+
+    private FileVisitor() {
+      this.pathMatcher = (p) -> true;
+    }
+
+    private FileVisitor(String pattern) {
+      if (pattern.startsWith("/")) {
```

Comment:
Not sure about this: If people pass in `*.java`, I think they should get only Java files in the current directory, not in all directories. In addition, when implementing it like this, we might need to check if we also need to add a separate check for Windows, right? WDYT?

Reply:
Ah, your right :/ so we need to be more careful with relative paths than what I was. And yes this is not compatible on Windows and that's a problem. I think we might be able to replace it with Paths.get(...).isAbsoluteParh().

Reply:
We could change it like this:
```java
    private FileVisitor(String pattern) {
      if (Paths.get(pattern).isAbsolute()) {
        this.pathMatcher = FileSystems.getDefault().getPathMatcher("glob:" + pattern);
      } else {
        this.pathMatcher = FileSystems.getDefault().getPathMatcher("glob:**/" + pattern);
      }
    }
```
But then there's still the problem about always iterating recursively. Because when `/home/user` is the current directory, `/home/user/*.java` and `*.java` would lead to different results, so I think it might be fine just to do
```java
    private FileVisitor(String pattern) {
      this.pathMatcher = FileSystems.getDefault().getPathMatcher("glob:**/" + pattern);
    }
```
WDYT?

Reply:
I now added a check for both relative _and_ absolute file, because when an absolute pattern is passed in, the relative file will not match!